scripts/core/superdesk-api.d.ts

```diff
@@ -694,7 +698,20 @@ declare module 'superdesk-api' {
         'content:update': IWebsocketMessage<IArticleUpdateEvent>;
     }
 
+    export interface IDeployConfig {
+        config?: {
+            planning_check_for_assignment_on_publish?: boolean;
```

Comment:
Can you move planning related types to planning extension?

You could type it like this:

```
// core
interface IDeployConfigMain {
    attachments_max_size: any;
    attachments_max_files: any;
    ingest_expiry_minutes: any;
    content_expiry_minutes: any;
}

interface IDeployConfig {
    config?: IDeployConfigMain;
}

// inside planning extension
interface IPlanningConfigMain extends IDeployConfigMain  {
    planning_check_for_assignment_on_publish?: boolean;
    planning_link_updates_to_coverage?: boolean;
}

interface IDeployConfigPlanning {
    config?: IPlanningConfigMain;
}
```